doc/panes/editor.rst

```diff
@@ -29,32 +31,264 @@ The Editor pane consists of the following areas:
+
+Python is supported out of the box, and advanced users can add completion and linting support for other languages by setting up LSP servers for them under :menuselection:`Preferences --> Completion and Linting --> Other languages`.
+
+
+Code completion
+~~~~~~~~~~~~~~~
+
+Automatic code completion as you type is enabled by default in the Editor, and can also be triggered manually with :kbd:`Ctrl-Space`/:kbd:`Cmd-Space`, showing you help, possible completions, and available code snippets.
+For example, typing ``cla`` will display the keyword ``class``, the decorator ``classmethod`` and two built-in snippets with class templates.
+Select the desired completion with the arrow keys and :kbd:`Enter`, or by double clicking.
+
+.. image:: /images/editor/editor-code-completion.png
+   :alt: Spyder's Editor pane, showing a code completion example
+
+You can enable or disable on-the-fly code completion, as well as modify when it is triggered and what results are shown, under :menuselection:`Preferences --> Completion and Linting --> General --> Completions`.
+Spyder also allows you to define custom completion snippets to use, in addition to the ones offered by LSP, under :menuselection:`Preferences --> Completion and Linting --> Advanced`.
+
+
+Linting and code style
+~~~~~~~~~~~~~~~~~~~~~~
+
+Spyder can optionally highlight syntax errors, style issues, and other potential problems with your code in the Editor, which can help you spot bugs quickly and make your code easier to read and understand.
+
+.. image:: /images/editor/editor-pane-code-error.png
+   :alt: Spyder's Editor pane, showing an example of a highlighted code error
+
+The Editor's basic linting, powered by `Pyflakes <https://github.com/PyCQA/pyflakes>`_, warns of syntax errors and likely bugs in your code.
+It is on by default, and can be disabled or customized under :menuselection:`Preferences --> Completion and Linting --> Linting`.
+
+.. image:: /images/editor/editor-linting-setting.png
+   :alt: Spyder's preferences dialog, showing linting settings
+
+Code style analysis, powered by `Pycodestyle <https://pycodestyle.pycqa.org/en/stable/>`_, flags deviations from the style conventions in :pep:`8`.
+It is not active by default; you can enable it and customize the `pycodestyle error codes <https://pycodestyle.pycqa.org/en/stable/intro.html#error-codes>`_ shown with the options under :menuselection:`Preferences --> Completion and Linting --> Code style and formatting --> Code Style`.
+
+.. image:: /images/editor/editor-code-style-setting.png
+   :alt: Spyder's preferences dialog, showing code style and formatting settings
+
+
+Introspection features
+~~~~~~~~~~~~~~~~~~~~~~
+
+If there's a function, class or variable that you want to jump to the definition of, :kbd:`Ctrl`/:kbd:`Cmd`-click its name in the Editor (or click its name and press :kbd:`Ctrl-G` / :kbd:`Cmd-G` to jump to the file and line where it is defined.
+
+.. image:: /images/editor/editor-go-to-definition.gif
```

Comment:
Could you hold on/freeze-frame the final result for maybe 1-2 seconds longer? Right now, it only flashes on screen for perhaps half a second or less, which makes it hard to see that it takes the user to the definition of the selected function. Thanks!